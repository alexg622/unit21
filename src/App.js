import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from './store'
import './App.css';
import Header from './components/Header'
import Repo from "./components/Index"
import Issue from "./components/Issue"
import {getUserRepos} from './actions/githubActions'

class App extends Component {
  render() {
   return (
     <Provider store={ store }>
       <Header />
       <Router>
         <div className="App">
           <Route exact path="/" component={ Repo } />
         </div>
       </Router>
     </Provider>
   );
 }
}


export default App;
