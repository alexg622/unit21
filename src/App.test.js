import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import App from './App';
import Enzyme, {mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe("Header component", () => {
  it("matches the snapshot component", () => {
    const tree = renderer.create(<App />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("says Github Manager", () => {
    const wrapper = mount(<App />)
    const text = wrapper.find(".header-title").text()
    expect(text).toEqual("Github Manager")
  })
})
