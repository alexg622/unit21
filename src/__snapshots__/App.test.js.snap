// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches the snapshot component 1`] = `
Array [
  <div
    className="header-container"
  >
    <div
      className="left-space-holder"
    />
    <div
      className="header-title"
    >
      Github Manager
    </div>
    <form
      className="api-key-form"
      onSubmit={[Function]}
    >
      <input
        className="api-key-input"
        onChange={[Function]}
        placeholder="Github api key"
        type="text"
        value=""
      />
      <input
        className="api-key-submit"
        type="submit"
        value="Set Key"
      />
    </form>
  </div>,
  <div
    className="App"
  >
    <div
      className="index-container"
    >
      <div
        className="index-inner-container"
      >
        <div
          className="index-upper-container"
        >
          Repositories
        </div>
        <div
          className="index-lower-container"
        >
          <div
            className="instructions"
          >
            Enter your github auth token to view your repositories
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
